chef/cookbooks/monasca/recipes/agent.rb

```diff
@@ -34,8 +34,109 @@
 #   crowbar_framework/app/models/monasca_service.rb
 # * use credentials generated in the previous step to create a keystone account
 #   here (using keystone_register)
+# * Add agents_credentials field in template-monasaca.{json,schema}
+# * Second data structure named monasca_roles_cleanup that contains node
+#   accounts that are no longer needed
 
-agent_settings = node[:monasca][:agent][:config]
+agent_settings = node[:monasca][:agent]
+log ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> #{agent_settings.inspect}"
+
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
+
+monasca_port = node[:monasca][:api][:bind_port]
+ssl_enabled = node[:monasca][:api][:ssl]
+monasca_protocol = ssl_enabled ? "https" : "http"
+ha_enabled = node[:monasca][:ha][:enabled]
+
+my_admin_host = CrowbarHelper.get_host_for_admin_url(node, ha_enabled)
+my_public_host = CrowbarHelper.get_host_for_public_url(
+  node, ssl_enabled, ha_enabled
+)
+
+register_auth_hash = {
+  user: keystone_settings["admin_user"],
+  password: keystone_settings["admin_password"],
+  tenant: keystone_settings["admin_tenant"]
+}
+
+keystone_register "monasca-agent server wakeup keystone" do
+  protocol keystone_settings["protocol"]
+  insecure keystone_settings["insecure"]
+  host keystone_settings["internal_url_host"]
+  port keystone_settings["admin_port"]
+  auth register_auth_hash
+  action :wakeup
+end
+
+keystone_register "register monasca-agent user" do
+  protocol keystone_settings["protocol"]
+  insecure keystone_settings["insecure"]
+  host keystone_settings["internal_url_host"]
+  port keystone_settings["admin_port"]
+  auth register_auth_hash
+  user_name keystone_settings["service_user"]
+  user_password keystone_settings["service_password"]
+  tenant_name keystone_settings["service_tenant"]
+  action :add_user
+end
+
+keystone_register "give monasca-agent user access" do
+  protocol keystone_settings["protocol"]
+  insecure keystone_settings["insecure"]
+  host keystone_settings["internal_url_host"]
+  port keystone_settings["admin_port"]
+  auth register_auth_hash
+  user_name keystone_settings["service_user"]
+  tenant_name keystone_settings["service_tenant"]
+  role_name "admin"
+  action :add_access
+end
+
+keystone_register "register monasca-agent service" do
+  protocol keystone_settings["protocol"]
+  insecure keystone_settings["insecure"]
+  host keystone_settings["internal_url_host"]
+  port keystone_settings["admin_port"]
+  auth register_auth_hash
+  service_name "monasca-agent"
+  service_type "monitoring"
+  service_description "Monasca agent monitoring service"
+  action :add_service
+end
+
+keystone_register "register monasca-agent endpoint" do
```

Comment:
I think that needs to go api.rb, you install monasca-api you register an endpoint for it.